__tests__/common/tibybird.ts

```diff
@@ -0,0 +1,206 @@
+        json: async () => {
+          return {
+            data: [
+              {
+                foo: 'bar',
+              },
+            ],
+            rows: 1,
+          };
+        },
+      } as Response;
+    };
+
+    const client = new TinybirdClient(
+      mockAccessToken,
+      mockHost,
+      mockFetch as unknown as fetchfn,
+    );
+
+    const result = await client.query(mockQuery);
+    expect(result.rows).toEqual(1);
+    expect(result.data).toEqual([
+      {
+        foo: 'bar',
+      },
+    ]);
+  });
+
+  it('query error', async () => {
+    const mockFetch = async (): Promise<Response> => {
+      return {
+        ok: false,
+        text: async (): Promise<string> => {
+          return 'oops!';
+        },
+        status: 500,
+      } as Response;
+    };
+
+    const client = new TinybirdClient(
+      mockAccessToken,
+      mockHost,
+      mockFetch as unknown as fetchfn,
+    );
+
+    await expect(async () => {
+      await client.query(mockQuery);
+    }).rejects.toThrow('tinybird response 500: oops!');
+  });
+
+  const mockDatasource = 'datasouce';
+  const mockCsv = 'csv content';
+  const expectedMode = TinybirdDatasourceMode.APPEND;
+
+  it('postToDatasource success', async () => {
+    const mockFetch = async (
+      url: RequestInfo,
+      init?: RequestInit,
+    ): Promise<Response> => {
+      expect(url).toEqual(
+        'https://tinybird.co/v0/datasources?name=datasouce&mode=append',
+      );
+      expect(init).toEqual({
+        headers: {
+          Authorization: 'Bearer token',
+        },
+        method: 'POST',
+        body: mockCsv,
+      });
+
+      return {
+        ok: true,
+        json: async () => {
+          return {
+            data: 'dummy data',
+          };
+        },
+      } as Response;
+    };
+
+    const client = new TinybirdClient(
+      mockAccessToken,
+      mockHost,
+      mockFetch as unknown as fetchfn,
+    );
+
+    const result = await client.postToDatasource(
+      mockDatasource,
+      expectedMode,
+      mockCsv,
+    );
+
+    expect(result).toEqual({
+      data: 'dummy data',
+    });
+  });
+
+  it('postToDatasource error', async () => {
+    const mockFetch = async (): Promise<Response> => {
+      return {
+        ok: false,
+        text: async (): Promise<string> => {
+          return 'oops!';
+        },
+        status: 500,
+      } as Response;
+    };
+
+    const client = new TinybirdClient(
+      mockAccessToken,
+      mockHost,
+      mockFetch as unknown as fetchfn,
+    );
+
+    await expect(async () => {
+      await client.postToDatasource(mockDatasource, expectedMode, mockCsv);
+    }).rejects.toThrow('tinybird response 500: oops!');
+  });
+
+  it('Json2Csv date', async () => {
+    const records = [
+      {
+        column1: 'test',
+        column2: new Date('2023-07-27T16:47:33+0000'),
+      },
+    ];
+
+    const got = await TinybirdClient.json2Csv(records);
+    expect(got).toMatchSnapshot();
```

Comment:
Why did you decide on snapshot here?
It doesn't really matter but since it's a small value maybe inline would be best, to avoid snapshot issues down the line?

(Same goes for the other snapshots below here)